src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpAuthority.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+
+    internal sealed class HttpAuthority : IEquatable<HttpAuthority>
+    {
+        // ALPN Protocol Name should also be part of an authority, but we are special-casing for HTTP/3, so this can be assumed to be "H3".
+
+        public string IdnHost { get; }
+        public int Port { get; }
+
+        public HttpAuthority(string host, int port)
+        {
+            Debug.Assert(host != null);
+
+            // This is very rarely called, but could be optimized to avoid the URI-specific stuff by bringing in DomainNameHelpers from System.Private.Uri.
+
+            UriBuilder builder = new UriBuilder();
```

Comment:
Nit can use `var` here.
```suggestion
            var builder = new UriBuilder();
```